Modules/TestQuestionPool/classes/class.ilAssQuestionHintList.php

```diff
@@ -19,102 +19,53 @@
 /**
  * Model class for managing lists of hints for a question
  *
- * @author		Björn Heyser <bheyser@databay.de>
- * @version		$Id$
+ * @author         Björn Heyser <bheyser@databay.de>
+ * @version        $Id$
  *
- * @package		Modules/TestQuestionPool
+ * @package        Modules/TestQuestionPool
+ * @implements Iterator<ilAssQuestionHint>
  */
 class ilAssQuestionHintList implements Iterator
 {
-    /**
-     * the hint items array
-     *
-     * @access	private
-     * @var		array
-     */
-    private $questionHints = array();
+    /** @var list<ilAssQuestionHint> */
+    private array $questionHints = [];
 
-    /**
-     * iterator interface method
-     *
-     * @access	public
-     * @return	mixed
-     */
     public function current()
     {
         return current($this->questionHints);
     }
 
-    /**
-     * iterator interface method
-     *
-     * @access	public
-     * @return	mixed
-     */
-    public function rewind()
+    public function rewind(): void
     {
-        return reset($this->questionHints);
+        reset($this->questionHints);
     }
 
-    /**
-     * iterator interface method
-     *
-     * @access	public
-     * @return	mixed
-     */
-    public function next()
+    public function next(): void
     {
-        return next($this->questionHints);
+        next($this->questionHints);
     }
 
-    /**
-     * iterator interface method
-     *
-     * @access	public
-     * @return	mixed
-     */
-    public function key()
+    public function key(): int
     {
         return key($this->questionHints);
```

Comment:
Is this correct? Per docs, key returns int|string|null

Reply:
You can add a precise type in derivatives